### PR TITLE
feat(tags): adds tag scopes and model quality tagging

### PR DESCRIPTION
**Pull request by Encord-davids**

### CODE REVIEW

File: src/encord_active/app/common/components/bulk_tagging_form.py [hunk 1]
```diff
@@ -1,11 +1,19 @@
 from enum import Enum
+from time import perf_counter
```
Comment (frederik-encord) [inline src/encord_active/app/common/components/bulk_tagging_form.py:2]:
```suggestion
```

File: src/encord_active/app/common/components/bulk_tagging_form.py [hunk 2]
```diff
@@ -20,37 +28,43 @@ class BulkLevel(str, Enum):
...
 
-    for identifier, tags in all_df.loc[subset.identifier, "tags"].items():
-        if action == TagAction.ADD:
-            next = list(set(tags + selected_tags))
-        elif action == TagAction.REMOVE:
-            next = list(set(tag for tag in tags if tag not in selected_tags))
-        else:
-            raise Exception(f"Action {action} is not supported")
+    for tag in selected_tags:
+        start = perf_counter()
```
Comment (frederik-encord) [inline src/encord_active/app/common/components/bulk_tagging_form.py:43]:
```suggestion
```

File: src/encord_active/app/common/components/bulk_tagging_form.py [hunk 3]
```diff
@@ -20,37 +28,43 @@ class BulkLevel(str, Enum):
...
+        target_ids = [target_identifier(id, tag.scope) for id in subset.identifier.to_list()]
+        print(perf_counter() - start)
```
Comment (frederik-encord) [inline src/encord_active/app/common/components/bulk_tagging_form.py:45]:
```suggestion
```

File: src/encord_active/app/data_quality/sub_pages/explorer.py [hunk 4]
```diff
@@ -259,13 +253,13 @@ def fill_data_quality_window(current_df: pd.DataFrame):
     else:
         cols: List = []
         similarity_expanders = []
-        for i, (row_no, row) in enumerate(paginated_subset.iterrows()):
+        for i, (_, row) in enumerate(paginated_subset.iterrows()):
```
Comment (frederik-encord) [inline src/encord_active/app/data_quality/sub_pages/explorer.py:256]: Is `i` and `_` not the same thing?

File: src/encord_active/app/data_quality/common.py [hunk 5]
```diff
@@ -35,7 +36,7 @@ def get_metric_operation_level(pth: Path) -> str:
 
 
 @st.experimental_memo
-def load_available_metrics(metric_type_selection: MetricType) -> List[MetricData]:
+def load_available_metrics(metric_type_selection: str) -> List[MetricData]:
```
Comment (frederik-encord) [inline src/encord_active/app/data_quality/common.py:39]: Any reason why this is a string and not the Enum?

File: src/encord_active/app/views/landing_page.py [hunk 6]
```diff
@@ -24,6 +24,8 @@ def landing_page():
     with right_col:
         st.markdown((Path(__file__).parent.parent / "static" / "landing-page-documentation.md").read_text())
 
+    load_merged_df()
```
Comment (frederik-encord) [inline src/encord_active/app/views/landing_page.py:27]: Is this necessary? Is it for caching purposes? Perhaps we should to it on a thread then?

**Review by frederik-encord — CHANGES_REQUESTED** [review]

File: src/encord_active/app/data_quality/sub_pages/explorer.py (hunk 4 above)
Comment (Encord-davids) [reply src/encord_active/app/data_quality/sub_pages/explorer.py:256]: They are, addressed in our big refactor 👍

File: src/encord_active/app/views/landing_page.py (hunk 6 above)
Comment (Encord-davids) [reply src/encord_active/app/views/landing_page.py:27]: This loads the datafame to the state with no computations if the DB exists.

File: src/encord_active/app/data_quality/common.py (hunk 5 above)
Comment (Encord-davids) [reply src/encord_active/app/data_quality/common.py:39]: `@st.experimental_memo` requires all arguments to be hashable. (should be addressed by the big refactor)

**Review by frederik-encord — APPROVED** [review]

**Pull request merged**